src/main/java/core/basesyntax/MyTask.java

```diff
@@ -13,7 +15,32 @@ public MyTask(int startPoint, int finishPoint) {
 
     @Override
     protected Long compute() {
-        // write your code here
-        return null;
+        if (finishPoint - startPoint > 10) {
+            List<RecursiveTask<Long>> recursiveTasks = new ArrayList<>(createSubTask());
+            for (RecursiveTask<Long> recursiveTask : recursiveTasks) {
+                recursiveTask.fork();
+            }
+            Long result = 0L;
+            for (RecursiveTask<Long> recursiveTask : recursiveTasks) {
+                result += recursiveTask.join();
+            }
+            return result;
+        } else {
+            Long result = 0L;
+            for (int i = startPoint; i < finishPoint; i++) {
+                result += i;
+            }
+            return result;
+        }
+    }
+
+    private List<RecursiveTask<Long>> createSubTask() {
+        List<RecursiveTask<Long>> recursiveActions = new ArrayList<>();
+        int difference = finishPoint - startPoint;
```

Comment:
do you need to calculate the difference here?

Reply:
yes, so as not to confuse